### PR TITLE
KEP-4193: beta promotion of ServiceAccountTokenNodeBinding feature gate

### PR DESCRIPTION
**Pull request by liggitt**
https://github.com/kubernetes/enhancements/issues/4193

Includes 1.31 updates for the ServiceAccountTokenNodeBinding feature gate portion of KEP-4193 (the only changes made in 1.31)

/sig auth
/cc @munnerz @ritazh

### CODE REVIEW

File: content/en/docs/reference/command-line-tools-reference/feature-gates/service-account-token-node-binding.md [hunk 1]
```diff
@@ -9,5 +9,8 @@ stages:
   - stage: alpha 
     defaultValue: false
     fromVersion: "1.29"
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/service-account-token-node-binding.md:11]: add `toVersion: "1.30"`

File: content/en/docs/reference/command-line-tools-reference/feature-gates/service-account-token-node-binding.md [hunk 2]
```diff
@@ -9,5 +9,8 @@ stages:
...
+  - stage: beta
+    defaultValue: true
+    fromVersion: "1.31"
 ---
 Controls whether the apiserver allows binding service account tokens to Node objects.
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/service-account-token-node-binding.md:16]: nit; we prefer following the style guide more as things reach beta
```suggestion
Controls whether the API server allows binding service account tokens to Node objects.
Enabling this feature gate allows that binding.
```

**Comment by Princesso** [comment]
Hello @liggitt ,

v1.31 Doc Lead here. This PR has been marked as as risk for Doc Freeze.

The Doc Reviewers have given feedback on it, however, an update is yet to be made on it, based on feedback.

Please note that the Doc freeze deadline is tomorrow, and if the PR is not merged by tomorrow, you will need to file an exception or this enhancement will be removed from this release. Thank you!

**Comment by liggitt** [comment]
comments addressed

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md [hunk 3]
```diff
@@ -184,12 +184,13 @@ ServiceAccount. You can request a specific token duration using the `--duration`
...
+{{< feature-state feature_gate_name="ServiceAccountTokenNodeBinding" >}}
+
 When the `ServiceAccountTokenNodeBinding` and `ServiceAccountTokenNodeBindingValidation`
-features are enabled and the `KUBECTL_NODE_BOUND_TOKENS` environment variable is set to `true`,
-it is possible to create a service account token that is directly bound to a `Node`:
+features are enabled, it is possible to create a service account token that is directly bound to a `Node`:
 
 ```shell
-KUBECTL_NODE_BOUND_TOKENS=true kubectl create token build-robot --bound-object-kind Node --bound-object-name node-001 --bound-object-uid 123...456
+kubectl create token build-robot --bound-object-kind Node --bound-object-name node-001 --bound-object-uid 123...456
```
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/configure-service-account.md:193]:
```suggestion
# Requires kubectl v1.31 or later
kubectl create token build-robot --bound-object-kind Node --bound-object-name node-001 --bound-object-uid 123...456
```
?

I'm OK to merge this without that detail though; we can iterate.

**Comment by sftim** [comment]
Code PRs are all merged.

/lgtm
/approve

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md (hunk 3 above)
Comment (liggitt) [reply content/en/docs/tasks/configure-pod-container/configure-service-account.md:193]: opened follow-up in https://github.com/kubernetes/website/pull/47316

**Pull request merged**